File: java-ladidol/src/com/feng/newline/hashtable/L349两个数组的交集.java
package com.feng.newline.hashtable;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

/**
 * @projectName: leetcode
 * @package: com.feng.newline.hashtable
 * @className: L349两个数组的交集
 * @author: Ladidol
 * @description: 取出两个数组中共有的元素，去重！
 * @date: 2022/7/18 13:23
 * @version: 1.0
 */
public class L349两个数组的交集 {
    class Solution {
        public int[] intersection(int[] nums1, int[] nums2) {
            boolean[] hashTable1 = new boolean[1000];
//            ArrayList<Integer> resultList = new ArrayList<>();//为了不重复要用set
            Set<Integer> resultSet = new HashSet<>();
            for (int i : nums1) {
                hashTable1[i] = true;
            }
            for (int i : nums2) {
                if (hashTable1[i]){
                    resultSet.add(i);
                }
            }
            int index = 0;
            int[] result = new int[resultSet.size()];
            //将结果几何转为数组
            for (int i : resultSet) {
                result[index++] = i;
            }
            return result;
        }
    }
}

File: golang-devildyw/simple/L349两个数组的交集.go
package main

// 给定两个数组 nums1 和 nums2 ，返回 它们的 交集 。输出结果中的每个元素一定是 唯一 的。我们可以 不考虑输出结果的顺序 。
func main() {

}

func intersection(nums1 []int, nums2 []int) []int {
	var (
		intMap map[int]int = make(map[int]int)
		result []int       = make([]int, 0)
	)

	for i := 0; i < len(nums1); i++ {
		intMap[nums1[i]] = i
	}

	for i := 0; i < len(nums2); i++ {
		if _, ok := intMap[nums2[i]]; ok {
			result = append(result, nums2[i])
			delete(intMap, nums2[i])
		}
	}
	return result
}
